packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowPagePagination.ts

```diff
@@ -0,0 +1,224 @@
+import { useRecoilValue } from 'recoil';
+
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { useObjectNamePluralFromSingular } from '@/object-metadata/hooks/useObjectNamePluralFromSingular';
+import { formatFieldMetadataItemsAsFilterDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions';
+import { formatFieldMetadataItemsAsSortDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsSortDefinitions';
+import { generateDepthOneRecordGqlFields } from '@/object-record/graphql/utils/generateDepthOneRecordGqlFields';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { turnSortsIntoOrderBy } from '@/object-record/object-sort-dropdown/utils/turnSortsIntoOrderBy';
+import { recordPositionInternalState } from '@/object-record/record-field/states/recordPositionInternalState';
+import { turnObjectDropdownFilterIntoQueryFilter } from '@/object-record/record-filter/utils/turnObjectDropdownFilterIntoQueryFilter';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { usePrefetchedData } from '@/prefetch/hooks/usePrefetchedData';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKey';
+import { View } from '@/views/types/View';
+import { mapViewFiltersToFilters } from '@/views/utils/mapViewFiltersToFilters';
+import { mapViewSortsToSorts } from '@/views/utils/mapViewSortsToSorts';
+import { isDefined } from '~/utils/isDefined';
+
+export const useRecordShowPagePagination = (
+  propsObjectNameSingular: string,
+  propsObjectRecordId: string,
+) => {
+  const {
+    objectNameSingular: paramObjectNameSingular,
+    objectRecordId: paramObjectRecordId,
+  } = useParams();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const viewIdQueryParam = searchParams.get('view');
+
+  const recordPositionInternal = useRecoilValue(recordPositionInternalState);
+
+  const [currentRecordIndex, setCurrentRecordIndex] = useState(0);
+  const [hasPreviousRecord, setHasPreviousRecord] = useState(false);
+  const [hasNextRecord, setHasNextRecord] = useState(false);
+  const [objectRecords, setObjectRecords] = useState<ObjectRecord[]>([]);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [isLoadedRecords, setIsLoadedRecords] = useState(false);
+  const [isLoadingPagination, setIsLoadingPagination] = useState(true);
+  const [hasNextPage, setHasNextPage] = useState<boolean | undefined>();
+
+  const [viewName, setViewName] = useState('');
+
+  const objectNameSingular = propsObjectNameSingular || paramObjectNameSingular;
+  const objectRecordId = propsObjectRecordId || paramObjectRecordId;
+
+  if (!objectNameSingular || !objectRecordId) {
+    throw new Error('Object name or Record id is not defined');
+  }
+
+  const { objectMetadataItem } = useObjectMetadataItem({ objectNameSingular });
+
+  const { objectNamePlural } = useObjectNamePluralFromSingular({
```

Comment:
not important but you already have it easily with `objectMetadataItem.namePlural` so maybe no need to use this method which is more costly?